crates/forge-runner/src/backtrace.rs

```diff
@@ -0,0 +1,224 @@
+use anyhow::{Context, Result};
+use cairo_annotations::annotations::coverage::{
+    CodeLocation, ColumnNumber, CoverageAnnotationsV1, LineNumber, VersionedCoverageAnnotations,
+};
+use cairo_annotations::annotations::profiler::{
+    FunctionName, ProfilerAnnotationsV1, VersionedProfilerAnnotations,
+};
+use cairo_annotations::annotations::TryFromDebugInfo;
+use cairo_lang_sierra::program::StatementIdx;
+use cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+use cairo_lang_starknet_classes::contract_class::ContractClass;
+use cheatnet::runtime_extensions::forge_runtime_extension::contracts_data::ContractsData;
+use cheatnet::state::EncounteredError;
+use indoc::indoc;
+use itertools::Itertools;
+use rayon::prelude::*;
+use starknet_api::core::ClassHash;
+use std::collections::HashMap;
+use std::fmt::Display;
+use std::{env, fmt};
+
+const BACKTRACE_ENV: &str = "SNFORGE_BACKTRACE";
+
+pub fn add_back_trace_footer(
```

Comment:
nit: Should be applied to other usages as well
```suggestion
pub fn add_backtrace_footer(
```